src/scene/graph-node.js

```diff
@@ -1206,30 +1196,28 @@ class GraphNode extends EventHandler {
      * this.entity.insertChild(e, 1);
      */
     insertChild(node, index) {
-        if (node._parent !== null)
-            throw new Error('GraphNode is already parented');
-
-        // #if _DEBUG
-        this._debugInsertChild(node);
-        // #endif
 
+        this._prepareInsertChild(node);
         this._children.splice(index, 0, node);
         this._onInsertChild(node);
     }
 
-    // #if _DEBUG
     /**
+     * Prepares node for being inserted to a parent node.
+     *
      * @param {GraphNode} node - The node being inserted.
      * @private
      */
-    _debugInsertChild(node) {
-        if (this === node)
-            throw new Error('GraphNode cannot be a child of itself');
+    _prepareInsertChild(node) {
+
+        Debug.assert(!node._parent, `GraphNode ${node?.name} is already parented`);
+        Debug.assert(node !== this, `GraphNode ${node?.name} cannot be a child of itself`);
+        Debug.assert(!this.isDescendantOf(node), `GraphNode ${node?.name} cannot add an ancestor as a child`);
 
-        if (this.isDescendantOf(node))
-            throw new Error('GraphNode cannot add an ancestor as a child');
+        // remove it from existing parent
+        if (node._parent)
+            node._parent.removeChild(node);
```

Comment:
If the new behavior is that a node is unparented if it already has a parent, perhaps this should be mentioned in the API ref docs?

Reply:
But this triggers a console error, so it's not expected / new behaviour. It's just a mitigation for the app to reasonably continue instead of crash / create invalid graph.

Reply:
Any reason it _can't_ just be documented behavior and not a mitigation?